webrtc/RTCPeerConnectionIceEvent-constructor.html

```diff
@@ -1,34 +1,148 @@
-  var e = new RTCPeerConnectionIceEvent('type', { candidate: c, url: 'url', bubbles: true, cancelable: true});
-  assert_equals(e.type, 'type');
-  assert_equals(e.candidate, c);
-  // assert_equals(e.url, 'url');
-  assert_true(e.bubbles);
-  assert_true(e.cancelable);
-}, 'RTCPeerConnectionIceEvent constructor with full arguments.');
+/*
+RTCPeerConnectionIceEvent
+
+[Constructor(DOMString type, optional RTCPeerConnectionIceEventInit eventInitDict)]
+
+interface RTCPeerConnectionIceEvent : Event {
+    readonly attribute RTCIceCandidate? candidate;
+    readonly attribute DOMString?       url;
+};
+ */
+test(() => {
+  assert_throws(new TypeError(), () => {
+    new RTCPeerConnectionIceEvent();
+  });
+}, "RTCPeerConnectionIceEvent with no arguments throws TypeError");
+
+test(() => {
+  const event = new RTCPeerConnectionIceEvent("type");
+  /*
+  candidate of type RTCIceCandidate, readonly, nullable
+  url of type DOMString, readonly, nullable
+  */
+  assert_equals(event.candidate, null);
+  assert_equals(event.url, null);
+
+  /*
+  Firing an RTCPeerConnectionIceEvent event named e with an RTCIceCandidate candidate means that an event with the name e, which does not bubble (except where otherwise stated) and is not cancelable (except where otherwise stated),
+  */
+  assert_false(event.bubbles);
+  assert_false(event.cancelable);
+
+}, "RTCPeerConnectionIceEvent with no eventInitDict (default)");
+
+test(() => {
+  const event = new RTCPeerConnectionIceEvent("type", {});
+
+  /*
+  candidate of type RTCIceCandidate, readonly, nullable
+  url of type DOMString, readonly, nullable
+  */
+  assert_equals(event.candidate, null);
+  assert_equals(event.url, null);
+
+  /*
+  Firing an RTCPeerConnectionIceEvent event named e with an RTCIceCandidate candidate means that an event with the name e, which does not bubble (except where otherwise stated) and is not cancelable (except where otherwise stated),
+  */
+  assert_false(event.bubbles);
+  assert_false(event.cancelable);
+
+}, "RTCPeerConnectionIceEvent with empty object as eventInitDict (default)");
+
+test(() => {
+  const event = new RTCPeerConnectionIceEvent("type", {
+    candidate: null
+  });
+  assert_equals(event.candidate, null);
+}, "RTCPeerConnectionIceEvent.candidate is null when constructed with { candidate: null }");
+
+test(() => {
+  const event = new RTCPeerConnectionIceEvent("type", {
+    candidate: undefined
+  });
+  assert_equals(event.candidate, null);
+}, "RTCPeerConnectionIceEvent.candidate is null when constructed with { candidate: undefined }");
+
+
+/*
+
+4.8.1 RTCIceCandidate Interface
+
+The RTCIceCandidate() constructor takes a dictionary argument, candidateInitDict, whose content is used to initialize the new RTCIceCandidate object. When run, if both the sdpMid and sdpMLineIndex dictionary members are null, throw a TypeError.
+*/
+let candidate = "";
+let sdpMid = "sdpMid";
+let sdpMLineIndex = 1;
+let ufrag = "";
+let url = "foo.bar";
```

Comment:
It could be clearer if the variables are declared `const` here. Individual tests that have different parameters could declare their own local variables instead of modifying a globally shared parameter, which could accidentally affect other tests if we are not careful.